Apps/WebClient/src/ClientApp/src/components/timeline/filters.vue

```diff
@@ -0,0 +1,277 @@
+    config!: WebClientConfiguration;
+    @Getter("isOpen", { namespace: "sidebar" }) isSidebarOpen!: boolean;
+
+    private logger: ILogger = container.get(SERVICE_IDENTIFIER.Logger);
+    private eventBus = EventBus;
+    private isVisible: boolean = false;
+    private selectedFilters: string[] = [];
+    private windowWidth: number = 0;
+    private filters: Filter[] = [
+        {
+            name: "medication",
+            value: "Medication",
+            display: "Medications",
+            isEnabled: false,
+        },
+        {
+            name: "immunization",
+            value: "Immunization",
+            display: "Immunizations",
+            isEnabled: false,
+        },
+        {
+            name: "laboratory",
+            value: "Laboratory",
+            display: "Laboratory",
+            isEnabled: false,
+        },
+        {
+            name: "note",
+            value: "Note",
+            display: "My Notes",
+            isEnabled: false,
+        },
+        {
+            name: "encounter",
+            value: "Encounter",
+            display: "MSP Visits",
+            isEnabled: false,
+        },
+    ];
+
+    private get isMobileView(): boolean {
+        return this.windowWidth < 576;
+    }
+
+    @Watch("isMobileView")
+    private onIsMobileView() {
+        this.isVisible = false;
+    }
+
+    @Watch("isSidebarOpen")
+    private onIsSidebarOpen(newValue: boolean, oldValue: boolean) {
+        this.isVisible = false;
+    }
+
+    @Watch("selectedFilters")
+    private onFilterUpdate() {
+        this.filtersChanged();
+    }
+
+    @Emit()
+    public filtersChanged() {
+        if (this.selectedFilters.length > 0) {
+            return this.selectedFilters;
+        } else {
+            return this.getAllFilters();
+        }
+    }
+
+    private created() {
+        window.addEventListener("resize", this.handleResize);
+        this.handleResize();
+    }
+
+    private mounted() {
+        this.filters[0].isEnabled = this.config.modules["Medication"];
+        this.filters[1].isEnabled = this.config.modules["Note"];
+        this.filters[2].isEnabled = this.config.modules["Immunization"];
+        this.filters[3].isEnabled = this.config.modules["Laboratory"];
+        this.filters[4].isEnabled = this.config.modules["Encounter"];
+        this.selectedFilters = [];
+
+        var self = this;
+        this.eventBus.$on("filterSelected", function (filterName: string) {
+            self.onExternalFilterSelection(filterName);
+        });
+    }
+
+    private destroyed() {
+        window.removeEventListener("handleResize", this.handleResize);
+    }
+
+    private handleResize() {
+        this.windowWidth = window.innerWidth;
+    }
+
+    private toggleMobileView() {
+        this.isVisible = !this.isVisible;
+    }
+
+    private clearFilters(): void {
+        this.selectedFilters = [];
+    }
+
+    private onExternalFilterSelection(filterName: string) {
```

Comment:
as this method updates the selectedFilters, can we call this updateSelectedFilter?

Reply:
the catch is that this is only called when from an external call via the bus. IMHO updateSelectedFilter gives the impression that this updates it in a general way

Reply:
Mbe an item for retro to define our naming convention as for me method should have a meaningful name and its name should tell what it does. Approve this for now then :-)